AEPIdentity/Sources/Identity.swift

```diff
@@ -55,41 +55,55 @@ import Foundation
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
-        guard canProcessEvents(event: event) else { return false }
+        guard let state = state else { return false }
+
+        // fast boot identity without waiting for configuration
+        if state.boot(event: event, createSharedState: createSharedState(data:event:)) {
+
+            // skip waiting for configuration if event is getExperienceCloudId event or getIdentifiers event
+            if event.isGetIdentifierEvent {
+                Log.trace(label: "\(name):\(#function)", "Not waiting for latest configuration as we have the ECID cached and available for processing [event:(\(event.name)) id:(\(event.id)].")
+                return true
+            }
+        }
+
+        guard let configurationSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value else {
+            Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state to get required configuration fields before processing [event:(\(event.name)) id:(\(event.id)].")
+            return false
```

Comment:
Here you are getting the latest Configuration state for each Identity event, but it is only needed when booting and performing the first force sync event. You could wrap this in an `if !state.hasSynced ...` block, but I think it would be cleaner to pass the `getSharedState` function to `IdentityState.forceSyncIdentifiers` so all the logic to validate the initial force sync event is handled there.